convert2rhel/checks.py

```diff
@@ -57,6 +62,88 @@ def perform_pre_ponr_checks():
     ensure_compatibility_of_kmods()
 
 
+def check_convert2rhel_latest():
+    """Make sure that we are running the latest version of convert2rhel"""
+
+    # get lastest avalible version
+
+    repo_content = (
+        "[convert2rhel]\n"
+        "name=Convert2RHEL Repository\n"
+        "baseurl=https://ftp.redhat.com/redhat/convert2rhel/$releasever/os/\n"
+        "gpgcheck=0\n"
+        "enabled=1\n"
+    )
+
+    repo_dir = os.path.join(utils.TMP_DIR, "convert2rhel_repo")
+    utils.mkdir_p(repo_dir)
+    # repo_dir = os.path.join("/tmp/","convert2rhel_repo")
+    repo_path = os.path.join(repo_dir, "convert2rhel.repo")
+    store_content_to_file(filename=repo_path, content=repo_content)
+
+    # repoquery --setopt=reposdir=$(pwd)/test_repo --releasever=8 convert2rhel
+    raw_output_convert2rhel_versions, _ = run_subprocess(
+        "repoquery --releasever={releasever} --setopt=reposdir={repodir} convert2rhel".format(
+            releasever=system_info.releasever,
+            # releasever="8",
+            repodir=repo_dir,
+        ),
+        print_output=False,
+    )
+
+    #   - if cannot access url: print warning and continue
+    PKG_NEVR = r"\b(\S+)-(?:([0-9]+):)?(\S+)-(\S+)\b"
+    convert2rhel_versions = re.findall(PKG_NEVR, raw_output_convert2rhel_versions, re.MULTILINE)
+
+    latest_version = ("0", "0.00", "0")
+    #   - get latest version from https://ftp.redhat.com/redhat/convert2rhel/.
+    for package_version in convert2rhel_versions:
+        if package_version[0] != "convert2rhel":
+            continue
+
+        ver_compare = rpm.labelCompare(package_version[1:], latest_version)
+
+    # if current version is not latest:
+    ver_compare = rpm.labelCompare(("0", convert2rhel_version, "0"), ("0", latest_version[1], "0"))
+    # - If envar is detected
+    if "CONVERT2RHEL_UNSUPPORTED_VERSION" in os.environ:
+        logger.warning(
+            "You are currently running %s, the latest version of convert2rhel is %s.\n"
+            "'CONVERT2RHEL_UNSUPPORTED_VERSION' environment detected, continuing conversion"
+            % (convert2rhel_version, latest_version)
+        )
+
+    # - If converting rhel6:
+    elif ver_compare == -1:
+        if system_info.version.major <= 6:
+            logger.critical(
+                "You are currently running %s, the latest version of convert2rhel is %s.\n"
+                "Only the latest version is supported for conversion. If you want to ignore"
+                " this you can set 'CONVERT2RHEL_UNSUPPORTED_VERSION' to continue"
+                % (convert2rhel_version, latest_version)
+            )
+
+        else:
+            logger.info(
+                "You are currently running %s, the latest version of convert2rhel is %s.\n"
+                "Continuing conversion." % (convert2rhel_version, latest_version)
+            )
```

Comment:
What do you think about changing this to be a simple warning to the user if they are on the RHEL6 since we don't officially support it? 

Maybe something like this: 

```suggestion
    if system_info.version.major <= 6:
        logger.warning(
            "You're running an unsupported operating system version for convert2rhel. "
            "If you want to ignore  this, then you can set 'CONVERT2RHEL_UNSUPPORTED_VERSION' to continue"
        )
```

The message is just a placeholder, feel free to improve it if you think it's a good change.